framework/include/utils/DependencyResolver.h

```diff
@@ -226,25 +262,59 @@ DependencyResolver<T>::insertDependency(const T & key, const T & value)
   if (_unique_deps.count(k) > 0)
     return;
   _unique_deps.insert(k);
+  auto insert_it = _depends.insert(k);
 
-  if (dependsOn(value, key))
+  for (auto & pr : _colors)
+    pr.second = WHITE;
+  _colors.emplace(key, WHITE);
+  _colors.emplace(value, WHITE);
+  if (hasCycle(key))
   {
     decltype(_depends) depends_copy(_depends);
-
-    // Insert the breaking dependency here so it will reflect properly in the exception
-    depends_copy.insert(k);
+    _depends.erase(insert_it);
 
     throw CyclicDependencyException<T>(
```

Comment:
@lindsayad @lingzou 
SAM is finding a new cyclic dependency, how do we return the name of the key here? It would help them
I imagine with the template _console << key or similar wont work since it's not defined. can we do some sort of template enable_if inside a routine? or should we duplicate the function ?